### PR TITLE
Newcomers_Guide.rst: Mention smart questions

### PR DESCRIPTION
**Pull request by 44rdvark**
Closes https://github.com/coala/coala/issues/4847

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [X] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [X] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by soumya-ranjan7** [comment]
ack 38f6a4c

**Comment by soumya-ranjan7** [comment]
ack 38f6a4c

**Comment by prashant0598** [comment]
unack 38f6a4c

**Comment by 44rdvark** [comment]
I initially didn't include the commit body because the [guide](http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html) specifically mentions it is not mandatory and I thought the shortlog for my commit was descriptive enough. What are the cases that warrant not writing the commit body then?

**Comment by 44rdvark** [comment]
Thank you for your answer, I've updated the pull request and hope it's fine now.

**Comment by nealkarpe** [comment]
@44rdvark Still shows two commits. Please squash them together.

**Comment by mridubhatnagar** [comment]
LGTM.
ack  f000f4e

File: docs/Developers/Newcomers_Guide.rst [hunk 1]
```diff
@@ -140,6 +140,12 @@ to be part of the Newcomer team. The invitation will be sent by mail and you
 will have to accept it to join. If you don't find the invitation, accept it
 `here <https://github.com/coala>`__.
 
+.. note::
```
Comment (Makman2) [inline docs/Developers/Newcomers_Guide.rst]: I would add an empty line here to conform to the style of other notes in this document

File: docs/Developers/Newcomers_Guide.rst [hunk 2]
```diff
@@ -140,6 +140,12 @@ to be part of the Newcomer team. The invitation will be sent by mail and you
...
+    If you have any questions, ask them in a precise and respectful way that
+    maximizes your chances of getting the answer you're looking for.
+    If you're unsure how to do that, please read: `How To Ask Questions
+    The Smart Way <http://www.catb.org/esr/faqs/smart-questions.html>`_.
```
Comment (Makman2) [inline docs/Developers/Newcomers_Guide.rst]: I would maybe put the note after the "Congratulations" paragraph below. What do you think?

**Comment by Makman2** [comment]
unack f000f4e

**Pull request closed without merging**